storage/core/src/main/java/com/hortonworks/registries/storage/impl/jdbc/JdbcStorageManager.java

```diff
@@ -102,6 +104,49 @@ public void update(Storable storable) {
         return entry;
     }
 
+    @Override
+    public boolean readLock(StorableKey key, Long time, TimeUnit timeUnit) {
+        log.debug("Obtaining a read lock for entry with storable key [{}]", key);
+
+        Supplier<Collection<Storable>> supplier = () -> queryExecutor.selectForShare(key);
+
+        try {
+            return getLock(supplier, time, timeUnit);
+        } catch (InterruptedException e) {
+            throw new StorageException("Failed to obtain a write lock for storable key : " + key);
+        }
+    }
+
+    @Override
+    public boolean writeLock(StorableKey key, Long time, TimeUnit timeUnit) {
+        log.debug("Obtaining a write lock for entry with storable key [{}]", key);
+
+        Supplier<Collection<Storable>> supplier = () -> queryExecutor.selectForUpdate(key);
+
+        try {
+            return getLock(supplier, time, timeUnit);
+        } catch (InterruptedException e) {
+            throw new StorageException("Failed to obtain a write lock for storable key : " + key);
+        }
+    }
+
+    private boolean getLock(Supplier<Collection<Storable>> supplier, Long time, TimeUnit timeUnit) throws InterruptedException {
+        Long remainingTime = TimeUnit.MILLISECONDS.convert(time, timeUnit);
```

Comment:
Declare `remainingTime` and `startTime` as primitives as boxing/unboxing are not required as all usages are primitive.